pingen/models/pingen.py

```diff
@@ -121,22 +96,31 @@ def _send(self, method, endpoint, **kwargs):
         :param str endpoint: endpoint to call
         :param kwargs: additional arguments forwarded to the requests method
         """
-        complete_url = urlparse.urljoin(self.url, endpoint)
 
-        response = method(complete_url, **kwargs)
+        p_url = urlparse.urljoin(self.url, endpoint)
+
+        if endpoint == 'document/get':
+            complete_url = '{}{}{}{}{}'.format(p_url,
+                                               '/id/',
+                                               kwargs['params']['id'],
+                                               '/token/',
+                                               self._token)
+        else:
+            complete_url = '{}{}{}'.format(p_url,
+                                           '/token/',
+                                           self._token)
 
-        if not response.ok:
-            raise ConnectionError(
-                "%s: %s" % (response.json['errorcode'],
-                            response.json['errormessage']))
```

Comment:
What is the rationale for removing this? I think you should at least use response.raise_for_status(), otherwise you might have no json at all if you receive an HTTP response with another status code than 200

Reply:
When a connection error really happens, i.e. no internet, a ConnectionError (from python lib requests.exceptions) will be raised by the previous instruction `response = method(complete_url, **kwargs)` and this code won't be executed at all. 
The logic was simply, if there's a response it's not a ConnectionError but an APIError instead.
Maybe I'm missing a use case ?

Reply:
In case the API key is wrong, here is the reponse we get :
```
(Pdb) response.json()
{u'errorcode': 99400, u'errormessage': u'Your token is invalid or expired', u'error': True}
```
So it will be catched as an API error and the message will be displayed correctly on pingen.document